core/src/main/scala/kafka/common/Topic.scala

```diff
@@ -24,7 +24,7 @@ import scala.collection.immutable
 object Topic {
 
   val GroupMetadataTopicName = "__consumer_offsets"
-  val TransactionStateTopicName = "__transaction_state"
+  val TransactionStateTopicName = "__transaction_state" // TODO: Should this be in the idempotent producer patch?
```

Comment:
I think it is fine, since the initPID part of TC needs to be in that patch; this not necessarily requires a txn log but since it already contains the skeleton of the TC already..